src/spikeinterface/core/template.py

```diff
@@ -0,0 +1,60 @@
+import json
+from dataclasses import dataclass, field
+
+import numpy as np
+
+from spikeinterface.core.sparsity import ChannelSparsity
+
+
+@dataclass
+class Templates:
+    templates_array: np.ndarray
+    sparsity: ChannelSparsity = None
+    num_units: int = field(init=False)
+    num_samples: int = field(init=False)
+    num_channels: int = field(init=False)
+
+    def __post_init__(self):
+        self.num_units, self.num_samples, self.num_channels = self.templates_array.shape
```

Comment:
How should we handle the templates_array when they are sparse, so that `sel.num_channels` is actually `self.max_num_sparse_channels`:

```
if self.sparsity_mask is None:
    self.num_units, self.num_samples, self.num_channels = self.templates_array.shape
else:
    self.num_units, self.num_samples, self.max_num_sparse_channels = self.templates_array.shape
```

Reply:
Done.